components/scream/src/control/surface_coupling.cpp

```diff
@@ -148,13 +148,13 @@ register_export (const std::string& fname,
   if (m_field_mgr->has_field(fname)) {
 
     // If fname is a field in the the field manager, then set the data and column info based on this field.
-    // Note that the field must not have a parent as get_view() would be unsafe.
+    // Note that the field must not have a parent as get_flattened_view() would be unsafe.
     export_field_type field = m_field_mgr->get_field(fname);
 
     EKAT_REQUIRE_MSG (field.is_allocated(), "Error! Export field view has not been allocated yet.\n");
 
     // Set view data ptr
-    info.data = field.get_view().data();
+    info.data = field.get_flattened_view().data();
```

Comment:
I believe that now when `qv` is registered in the surface coupling unit test this should throw since it is a subfield. The logic in surface coupling will need to change (currently we register fields with parents, and `get_col_info` checks if it is subfield and computes the correct stride).

Reply:
Looking at the testing log, this is exactly the test that fails.